repobee_sanitizer/_syntax.py

```diff
@@ -62,55 +65,50 @@ def check_syntax(lines: List[str]) -> None:
         lines: List containing every line of a text file as one element in the
             list.
     Raises:
-        plug.PlugError: Invalid Syntax.
+        A list of named tuples containing all found errors and their line
+            number.
     """
     last = Markers.END
     prefix = ""
     has_blocks = lines and contained_marker(lines[0]) == Markers.SHRED
 
     errors = _check_shred_syntax(lines)
     if errors:
-        raise plug.PlugError(errors)
+        return errors
 
     for line_number, line in enumerate(lines, start=1):
+
+        def _error(msg):
+            errors.append(Error(line_number, msg))
+
         marker = contained_marker(line)
         if marker == Markers.START:
             has_blocks = True
             if last != Markers.END:
-                errors.append(
-                    f"Line {line_number}: "
-                    "START block must begin file or follow an END block"
-                )
+                _error("START block must begin file or follow an END block")
             prefix = re.match(rf"(.*?){Markers.START.value}", line).group(1)
             last = Markers.START
         elif marker == Markers.REPLACE:
             if last != Markers.START:
-                errors.append(
-                    f"Line {line_number}: "
-                    "REPLACE-WITH block must follow START block"
-                )
+                _error("REPLACE-WITH block must follow START block")
             last = Markers.REPLACE
         elif marker == Markers.END:
             if last not in [Markers.START, Markers.REPLACE]:
-                errors.append(
-                    f"Line {line_number}: "
-                    "END block must follow START or REPLACE block"
-                )
+                _error("END block must follow START or REPLACE block")
             last = Markers.END
 
         if (
             last == Markers.REPLACE or marker == Markers.END
         ) and not line.startswith(prefix):
-            errors.append(f"Line {line_number}: Missing prefix")
+            _error(f"Missing prefix: {prefix}")
 
     if last != Markers.END:
-        errors.append("Final block must be an END block")
+        errors.append(Error(None, "Global: Final block must be an END block"))
```

Comment:
The `Global: ` part should be handled in the output formatting, now you are repeating it here in multiple places. In the output formatting, if you encounter a line that's `None`, put `Global` instead of `Line x`.